blog_post_collaboration/spark_streaming_dataset.md

```diff
@@ -0,0 +1,367 @@
+
+The following log will be shown once Kedro has run through all the nodes and the full Spark execution plan has been created. This doesn't mean the stream processing has stopped as the post run hook keeps the Spark Session alive. 
+
+```
+INFO Pipeline execution completed successfully. runner.py:93
+```
+
+As new data comes in, new Spark logs will be shown, even after the "Pipeline execution completed" log.
+
+```
+[Stage 22:> (0 + 1) / 1]
+[Stage 22:> (0 + 1) / 1][Stage 27:> (0 + 7) / 200]
+[Stage 22:> (0 + 1) / 1][Stage 27:> (1 + 6) / 200]
+[Stage 22:> (0 + 1) / 1][Stage 27:> (7 + 7) / 200]
+[Stage 22:> (0 + 1) / 1][Stage 27:> (11 + 7) / 200]
+[Stage 22:> (0 + 1) / 1][Stage 27:=> (14 + 7) / 200]
+```
+
+If there is an error in the input data, the Spark error logs will come through and Kedro will shut down the SparkContext and all the streams within it.
+
+```
+Caused by: java.lang.IllegalStateException: Cannot call methods on a stopped SparkContext.
+This stopped SparkContext was created at:
+org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
+sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
+sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
+sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
+java.lang.reflect.Constructor.newInstance(Constructor.java:423)
+py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
+py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
+py4j.Gateway.invoke(Gateway.java:238)
+py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
+py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
+py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
+py4j.ClientServerConnection.run(ClientServerConnection.java:106)
+java.lang.Thread.run(Thread.java:750)
+
+```
+
+
+
+
+## In summary
+
+To enable Kedro to work with Spark Structured Streaming, we developed a new Spark Streaming Dataset, as the existing Kedro Spark dataset was not compatible with Spark Streaming use cases. The new dataset was equipped with the following features to ensure seamless streaming:
+
+* A checkpoint location specification to avoid data duplication in streaming use cases.
+* The use of `.start()` at the end of the `_save` method to initiate the stream.
+
+Separate Hooks were added to the Kedro project to enable it to function as a streaming application. 
+
+Takeaways for the reader:
+
+* Kedro is easily extensible and you can do cool stuff as this team demonstrates with Spark Streaming integration
+
+However, you cannot at present viably use Kedro for your streaming workflows. Specifically for Spark, being able to deploy a streaming application is predicated on being able to deploy a job to a cluster, which is something that has been requested for a while but isn't support AFAIK. (The alternative may be to support cluster deployment.)
```

Comment:
We had a look on the original issue with YARN cluster mode, I think it is no more the case, because in this latest article https://kedro.org/blog/how-to-deploy-kedro-pipelines-on-amazon-emr , we mentioned how to deploy kedro pipeline on another application cluster.
In order to run spark streaming with the production workflow, as long as we spark-submit a job to have a running cluster with proper scheduler, it should work, an client project example was we used streaming on Databrick clusters.

Reply:
That's really useful, thanks @tingtingQB. I've made a few edits here to the summary but could use a little extra help from you, Haris or Tom to refine what i've added, and also to smooth out the introduction and summary to add any further points you think a reader would find useful.

Reply:
cc https://github.com/kuriantom369 & https://github.com/harisqb

Reply:
cc @kuriantom369 & @harisqb (full urls don't work as mentions I think)